asyncscope.md

```diff
@@ -279,201 +352,541 @@ listening activity is bound in the scope of the loop, the lifetime of handling r
+    noexcept(noexcept(scope.nest(std::forward<Sender>(snd)))
+    -> decltype(scope.nest(std::forward<Sender>(snd)));
+
+template <class Scope, class Sender>
+concept async_scope =
+    sender<Sender> &&
+    requires(Scope&& scope, Sender&& snd) {
+      { nest(std::forward<Sender>(snd), std::forward<Scope>(scope)) } -> sender;
+    };
+
+template <sender Sender, async_scope<Sender> Scope, class Env = empty_env>
+  requires sender_to<Sender, @@_spawn-receiver_@@<Env>>
+void spawn(Sender&& snd, Scope&& scope, Env env = {});
+
+template <sender Sender, async_scope<Sender> Scope, class Env = empty_env>
+@@_future-sender-t_@@<Sender, Env> spawn_future(Sender&& snd, Scope&& scope, Env env = {});
+
 struct counting_scope {
-  using self_t = counting_scope; /*@@_exposition-only_@@*/
-  counting_scope();
-  ~counting_scope();
-  counting_scope(const self_t&) = delete;
-  counting_scope(self_t&&) = delete;
-  self_t& operator=(const self_t&) = delete;
-  self_t& operator=(self_t&&) = delete;
-
-  void
-  /*@@_implementation-defined_@@*/ /*@@_customization-point_@@*/(
-    decays_to<self_t> auto&&, spawn_t, sender_to<@@_spawn-receiver_@@> auto&& s) noexcept;
-
-  template <sender_to<@@_spawn-future-receiver_@@> S>
-  @@_spawn-future-sender_@@<S>
-  /*@@_implementation-defined_@@*/ /*@@_customization-point_@@*/(
-    decays_to<self_t> auto&&, spawn_future_t, S&& s) noexcept;
-
-  template <sender S>
-  @@_nest-sender_@@<S>
-  /*@@_implementation-defined_@@*/ /*@@_customization-point_@@*/(
-    decays_to<self_t> auto&&, nest_t, S&& s) noexcept;
+    counting_scope() noexcept;
+    ~counting_scope();
+
+    // counting_scope is immovable and uncopyable
+    counting_scope(const counting_scope&) = delete;
+    counting_scope(counting_scope&&) = delete;
+    counting_scope& operator=(const counting_scope&) = delete;
+    counting_scope& operator=(counting_scope&&) = delete;
+
+    template <sender S>
+    struct @@_nest-sender_@@; // @@_exposition-only_@@
+
+    template <sender S>
+    [[nodiscard]] @@_nest-sender_@@<std::remove_cvref_t<S>> nest(S&& s) & noexcept(
+            std::is_nothrow_constructible_v<std::remove_cvref_t<S>, S>);
+
+    struct @@_join-sender_@@; // @@_exposition-only_@@
+
+    [[nodiscard]] @@_join-sender_@@ join() noexcept;
+
+    // observers in the spirit of std::weak_ptr<T>::expired() and
+    // std::shared_ptr<T>::use_count(); the values must be correct
+    // when computed but may be stale by the time they can be observed
+
+    [[nodiscard]] bool joined() const noexcept;
+
+    [[nodiscard]] bool join_started() const noexcept;
+
+    [[nodiscard]] size_t use_count() const noexcept;
 };
 ```
 
-## Lifetime
+## `execution::nest`
 
-The `counting_scope` keeps a counter of how many spawned _`async-function`_ s have not completed.
+```cpp
+template <sender Sender>
+auto nest(Sender&& snd, auto&& scope) noexcept(noexcept(scope.nest(std::forward<Sender>(snd)))
+    -> decltype(scope.nest(std::forward<Sender>(snd)));
```

Comment:
Actually, I take the oops back; `async_scope<Scope, Sender>` requires that `nest(sender, scope)` is valid so I expect that constraining `nest` with `async_scope` would lead to a recursive definition.